src/main/java/software/amazon/awssdk/crt/http/HttpStreamBaseResponseHandler.java

```diff
@@ -87,4 +87,12 @@ default int onResponseBody(HttpStreamBase stream, byte[] bodyBytesIn) {
      */
     void onResponseComplete(HttpStreamBase stream, int errorCode);
 
+    /**
+     * Called right before stream is complete, whether successful or unsuccessful.
+     * @param stream The HTTP stream to which the metrics apply
+     * @param metrics The [HttpStreamMetrics] containing metrics for the given stream
+     */
+    default void onMetrics(HttpStreamBase stream, HttpStreamMetrics metrics) {
+        /* Optional callback, nothing to do by default */
+    }
```

Comment:
nitpick: can we move the metrics callback before the complete callback? So, that it's the same order as it gets invoked